resources/role.rb

```diff
@@ -0,0 +1,152 @@
+property :role_name, String, name_property: true
+property :description, String, default: ''.freeze
+property :roles, Array, default: lazy { [] }
+property :privileges, Array, default: lazy { [] }
+property :api_endpoint, String, identity: true, default: lazy { node['nexus3']['api']['endpoint'] }
+property :api_username, String, identity: true, default: lazy { node['nexus3']['api']['username'] }
+property :api_password, String, identity: true, sensitive: true, default: lazy { node['nexus3']['api']['password'] }
+
+load_current_value do |desired|
+  api_endpoint desired.api_endpoint
+  api_username desired.api_username
+  api_password desired.api_password
```

Comment:
The 3 above lines are not needed because these properties are "identity"

Reply:
Done